openmc/deplete/pool.py

```diff
@@ -66,39 +65,45 @@ def deplete(func, chain, x, rates, dt, matrix_func=None, msr=None):
         matrices = map(matrix_func, repeat(chain), rates, fission_yields)
 
     if msr is not None:
-        # calculate removal rate terms as diagonal matrices
-        removal = map(chain.form_rr_term, repeat(msr), msr.burn_mats)
-        #subtract removal rate terms from Bateman matrices
-        matrices = [mat1 - mat2 for (mat1, mat2) in zip(matrices, removal)]
+        # Calculate removal rate terms as diagonal matrices
+        removals = map(chain.form_rr_term, repeat(msr), msr.burn_mats)
+        # Subtract removal rate terms from Bateman matrices
+        matrices = [matrix - removal for (matrix, removal) in zip(matrices, removals)]
 
         if len(msr.index_transfer) > 0:
-            # calculate transfer rate terms as diagonal matrices
-            transfer = list(map(chain.form_rr_term, repeat(msr),
+            # Calculate transfer rate terms as diagonal matrices
+            transfers = list(map(chain.form_rr_term, repeat(msr),
                                 msr.index_transfer))
 
             # Combine all matrices together in a single matrix of matrices
             # to be solved in one-go
             n_rows = n_cols = len(msr.burn_mats)
             rows = []
-            for row in range(n_rows):
-                cols = []
-                for col in range(n_cols):
-                    if row == col:
-                        # Fill the diagonals with the Bateman matrices
-                        cols.append(matrices[row])
-                    elif (msr.burn_mats[row], msr.burn_mats[col]) in \
-                                    msr.index_transfer:
-
-                        index = list(msr.index_transfer).index( \
-                                    (msr.burn_mats[row], msr.burn_mats[col]))
-                        # Fill the off-diagonals with the transfer matrices
-                        cols.append(transfer[index])
-                    else:
-                        cols.append(None)
-                rows.append(cols)
+            rows_cols = \
+                np.array(np.meshgrid(range(n_rows),
+                                     range(n_cols))).T.reshape(-1, 2)
```

Comment:
This is cool, but do we really need it?

Reply:
right, I've never actually tested for many many materials. Also, if we define many transfers (or feeding) between materials, the final matrix will become huge and cpu time might explode (or memory not allow it). It should probably be added as a note side in the theory manual, until some parallel solutions are implemented.

Reply:
Agreed